tests/Aspire.Hosting.Oracle.Tests/OracleFunctionalTests.cs

```diff
@@ -0,0 +1,377 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Components.Common.Tests;
+using Aspire.Hosting.Testing;
+using Aspire.Hosting.Utils;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Storage;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Polly;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Aspire.Hosting.Oracle.Tests;
+
+public class OracleFunctionalTests(ITestOutputHelper testOutputHelper)
+{
+    [Fact]
+    [RequiresDocker]
+    public async Task VerifyOracleResource()
+    {
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(10));
+        var pipeline = new ResiliencePipelineBuilder()
+            .AddRetry(new() { MaxRetryAttempts = 20, Delay = TimeSpan.FromSeconds(10) })
+            .Build();
+
+        var builder = CreateDistributedApplicationBuilder();
+
+        var oracleDbName = "freepdb1";
+
+        var oracle = builder.AddOracle("oracle");
+        var db = oracle.AddDatabase(oracleDbName);
+
+        using var app = builder.Build();
+
+        await app.StartAsync();
+
+        var hb = Host.CreateApplicationBuilder();
+
+        hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+        {
+            [$"ConnectionStrings:{db.Resource.Name}"] = await db.Resource.ConnectionStringExpression.GetValueAsync(default)
+        });
+
+        hb.AddOracleDatabaseDbContext<TestDbContext>(db.Resource.Name);
+
+        using var host = hb.Build();
+
+        await host.StartAsync();
+
+        // Wait until the database is available
+        await pipeline.ExecuteAsync(async token =>
+        {
+            var dbContext = host.Services.GetRequiredService<TestDbContext>();
+            var databaseCreator = (RelationalDatabaseCreator)dbContext.Database.GetService<IDatabaseCreator>();
+            Assert.True(await databaseCreator.CanConnectAsync(token));
+        }, cts.Token);
+
+        await pipeline.ExecuteAsync(async token =>
+        {
+            var dbContext = host.Services.GetRequiredService<TestDbContext>();
+
+            var results = await dbContext.Database.ExecuteSqlRawAsync("SELECT 1", token);
+
+            Assert.Equal(1, results);
+        }, cts.Token);
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false, Skip = "When using p@ssw0rd1 it fails with: Password cannot be null. Enter password: \n Then when using a different password the image takes 5 minutes to be ready.")]
```

Comment:
What does `p@ssw0rd1` have anything to do with this test?

Do we have an issue logged for this?

Reply:
https://github.com/dotnet/aspire/issues/5044